controllers/auth_config_controller.go

```diff
@@ -272,7 +284,15 @@ func (r *AuthConfigReconciler) translateAuthConfig(ctx context.Context, authConf
 			}
 
 		case api.PlainIdentityAuthentication:
-			translatedIdentity.Plain = &identity_evaluators.Plain{Pattern: identity.Plain.Selector}
+			if identity.Plain.Expression != "" {
+				expression, err := cel.NewStringExpression(string(identity.Plain.Expression))
```

Comment:
Not sure if understand `NewStringExpression` here. Does this mean `plain.expression` will always resolve to a string?

With `JSONValue`, I think `plain.selector` can resolve to any type, including returning complex objects to set `auth.identity`, no?

Reply:
So, what would be `plain.selector` value's type for `auth.identity` if one would be to select that? In Golang's type system that is...

Reply:
from gjson

```go
func (t Result) Value() interface{} {
	if t.Type == String {
		return t.Str
	}
	switch t.Type {
	default:
		return nil
	case False:
		return false
	case Number:
		return t.Num
	case JSON:
		r := t.arrayOrMap(0, true)
		if r.vc == '{' {
			return r.oi
		} else if r.vc == '[' {
			return r.ai
		}
		return nil
	case True:
		return true
	}
}
```

From gjson... So a `map[string]interface{}`, where `interface` is one of the above recursively I guess...

Reply:
One more question tho... what is that good for tho? I realize I took a shortcut here, and even left a note to make this better, but... we're just marshalling json back and forth, right? This only ever gets used thru `AuthPipeline.GetAuthorizationJSON`, no? i.e. _as JSON_?

Reply:
I think this is where the mismatch happens: 

```
	ResolveFor(jsonData string) (interface{}, error)
```

`ResolveFor` is the thing that only ever get to interact with these payloads, is that correct? So we end up spending all our CPU cycles marshaling that data back and forth, are we not? Am I missing something?

Reply:
> > So, what would be `plain.selector` value's type for `auth.identity` if one would be to select that? In Golang's type system that is...
> 
> Not sure if I follow. `plain.selector` is not for selecting _from_ `auth.identity`; it selects for setting the value _of_ `auth.identity`. What it selects can be of any type.

Not going to elaborate on this... as it's orthogonal and I found the answer, as [per this answer](https://github.com/google/cel-spec/blob/master/tests/simple/testdata/macros.textproto#L209-L213). But the follow up questions remain. I mean I can go thru the whole process of remapping to golang native types... to only marshal them back into JSON literals later... but it makes no sense, other than something _needing to use that in golang_. Tho even then I'd argue it might be better to optimize for the most frequent data access, which seems to be JSON based of my scanning of the code base. 

I do think that, if we ever 🔥 the gjson stuff, as per my comment in the code, ideally the pipeline stores all that data as protobuf... which would make it much lighter weight to inject it into the CEL interpreter. 

So _I can_ have the resulting value of a CEL expression be returned from `ResolveFor`, be expressed in golang's type system, but I really think this isn't sensible... as, again afaict! but I could be missing something, nothing actually needs that, other than to re marshal it back into JSON... 

Is my reasoning sensible? Am I missing something that renders it inapplicable to the problem space?

Reply:
Also, I think `ResolveFor(jsonData string) (interface{}, error)` should probably _not_ expect json in and golang out, but have a clear contract, i.e. `json string` in & out _or_ `interface{}` in & out... It's not really symmetrical. But _not arguing to fix this as part of this PR or even now neither_

Reply:
For future us... Not sensible for this PR I think

<details>

Thinking out loud further... 

`ResolveFor` should probably be `ResolveFor(pipeline *AuthPipeline) (interface{}, error)`. 
That now would let us control what the expression takes in as argument _format_. gjson stuff can call `GetAuthorizationJSON()`, while CEL could bypass it go straight to the protobuf format. I'd still argue for storing things in protobuf tho... So in order to remain "compatible", the result would be... "undefined" which probably isn't good... 🤔 

</details>

Reply:
Let me show I mean with an example...

Requisite:
1. Add the `omitempty` suggested at https://github.com/Kuadrant/authorino/pull/495#discussion_r1822380561

Run the service:

```sh
export LOG_LEVEL=debug; make cluster install run
```

Create a couple of AuthConfigs expected to behave exactly the same:

```sh
kubectl apply -f -<<EOF
apiVersion: authorino.kuadrant.io/v1beta3
kind: AuthConfig
metadata:
  name: gjson
spec:
  hosts: ["gjson"]
  authentication:
    "envoy-verified-jwt":
      plain:
        selector: metadata.filter_metadata.envoy\.filters\.http\.jwt_authn|verified_jwt
  authorization:
    "must_be_admin":
      patternMatching:
        patterns:
        - selector: auth.identity.roles
          operator: incl
          value: admin
---
apiVersion: authorino.kuadrant.io/v1beta3
kind: AuthConfig
metadata:
  name: cel
spec:
  hosts: ["cel"]
  authentication:
    "envoy-verified-jwt":
      plain:
        expression: |
          metadata.filter_metadata["envoy.filters.http.jwt_authn"].verified_jwt
  authorization:
    "must_be_admin":
      patternMatching:
        patterns:
        - predicate: |
            "admin" in auth.identity.roles
EOF
```

Send a request to the AuthConfig using `plain.selector`:

```sh
grpcurl -plaintext -d @ localhost:50051 envoy.service.auth.v3.Authorization.Check <<EOF
{
  "attributes": {
    "request": {
      "http": {
        "host": "gjson"
      }
    },
    "metadata_context": {
      "filter_metadata": {
        "envoy.filters.http.jwt_authn": {
          "verified_jwt": {
            "aud": "toystore",
            "exp": 1730284979,
            "iat": 1730371379,
            "iss": "auth-server.io",
            "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3",
            "roles": ["admin"]
          }
        }
      }
    }
  }
}
EOF
```
```
{
  "status": {},
  "okResponse": {},
  "dynamicMetadata": {}
}
```

Now send a request to the AuthConfig using `plain.expression`:

```sh
grpcurl -plaintext -d @ localhost:50051 envoy.service.auth.v3.Authorization.Check <<EOF
{
  "attributes": {
    "request": {
      "http": {
        "host": "cel"
      }
    },
    "metadata_context": {
      "filter_metadata": {
        "envoy.filters.http.jwt_authn": {
          "verified_jwt": {
            "aud": "toystore",
            "exp": 1730284979,
            "iat": 1730371379,
            "iss": "auth-server.io",
            "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3",
            "roles": ["admin"]
          }
        }
      }
    }
  }
}
EOF
```
```
{
  "status": {
    "code": 7
  },
  "deniedResponse": {
    "status": {
      "code": "Forbidden"
    },
    "headers": [
      {
        "header": {
          "key": "X-Ext-Auth-Reason",
          "value": "no such key: roles"
        }
      }
    ]
  }
}
```

If you check the logs, both requests succeed in the authentication, however...

The first request evaluates authorisation based on the following Authorization JSON:

```json
{
  "auth": {
    "identity": {
      "aud": "toystore",
      "exp": 1730284979,
      "iat": 1730371379,
      "iss": "auth-server.io",
      "roles": [
        "admin"
      ],
      "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3"
    }
  },
  "context": {
    "metadata_context": {
      "filter_metadata": {
        "envoy.filters.http.jwt_authn": {
          "verified_jwt": {
            "aud": "toystore",
            "exp": 1730284979,
            "iat": 1730371379,
            "iss": "auth-server.io",
            "roles": [
              "admin"
            ],
            "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3"
          }
        }
      }
    },
    "request": {
      "http": {
        "host": "gjson",
        "id": "b59827c5-ec62-422c-972c-b86f1d29ec52"
      }
    }
  },
  "destination": {},
  "metadata": {
    "filter_metadata": {
      "envoy.filters.http.jwt_authn": {
        "verified_jwt": {
          "aud": "toystore",
          "exp": 1730284979,
          "iat": 1730371379,
          "iss": "auth-server.io",
          "roles": [
            "admin"
          ],
          "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3"
        }
      }
    }
  },
  "request": {
    "host": "gjson",
    "id": "b59827c5-ec62-422c-972c-b86f1d29ec52"
  },
  "source": {}
}
```

The second request evaluates authorisation based on the following Authorization JSON:

```json
{
  "auth": {
    "identity": "{\"aud\":\"toystore\", \"exp\":1730284979, \"iat\":1730371379, \"iss\":\"auth-server.io\", \"roles\":[\"admin\"], \"sub\":\"55474041-c8c6-48bb-9398-34c2dbfd88b3\"}"
  },
  "context": {
    "metadata_context": {
      "filter_metadata": {
        "envoy.filters.http.jwt_authn": {
          "verified_jwt": {
            "aud": "toystore",
            "exp": 1730284979,
            "iat": 1730371379,
            "iss": "auth-server.io",
            "roles": [
              "admin"
            ],
            "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3"
          }
        }
      }
    },
    "request": {
      "http": {
        "host": "cel",
        "id": "04eb5f71-ec3c-40f1-a941-a7a5f0b3390b"
      }
    }
  },
  "destination": {},
  "metadata": {
    "filter_metadata": {
      "envoy.filters.http.jwt_authn": {
        "verified_jwt": {
          "aud": "toystore",
          "exp": 1730284979,
          "iat": 1730371379,
          "iss": "auth-server.io",
          "roles": [
            "admin"
          ],
          "sub": "55474041-c8c6-48bb-9398-34c2dbfd88b3"
        }
      }
    }
  },
  "request": {
    "host": "cel",
    "id": "04eb5f71-ec3c-40f1-a941-a7a5f0b3390b"
  },
  "source": {}
}
```